.github/workflows/test-e2e.yaml

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The test-e2e action does not check out the repo. Instead, it triggers the e2e
+# Cloud Build using this action's commit SHA, then waits for the cloud build
+# run to complete.
+#
+# This action uses a Cloud Build trigger rather than `gcloud build submit`
+# because we want Cloud Build to check out the repo. We don't want to upload
+# the source from this github action.
+#
+# We don't want e2e build log to be printed in this Github Action's log
+# because it may contain secrets.
+#
+name: e2e-test-trigger
+on:
+  pull_request:
+  pull_request_target:
+    types: [labeled]
+jobs:
+  build:
+    if: "${{ github.event.action != 'labeled' || github.event.label.name == 'e2e: run' }}"
+    name: Trigger E2E test build in Google Cloud Build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - name: Remove PR Label
+        if: "${{ github.event.action == 'labeled' && github.event.label.name == 'e2e: run' }}"
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            try {
+              await github.rest.issues.removeLabel({
+                name: 'e2e: run',
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: context.payload.pull_request.number
+              });
+            } catch (e) {
+              console.log('Failed to remove label. Another job may have already removed it!');
+            }
+      - id: 'auth'
```

Comment:
as per the `auth` docs I think we need the checkout action to run prior to the auth action?

Reply:
https://github.com/google-github-actions/auth#prerequisites

Reply:
It works without the checkout step, however I have to add 'create_credentials_file: true'. or it puts a nasty warning in the log.

Reply:
This makes me think that maybe something is up, `create_credentials_file` is true by default  so setting it should cause no difference? It is not supposed to work without the checkout step I believe.